docs/source/models.rst

```diff
@@ -182,11 +189,172 @@ TensorFlow (``tensorflow``)
+popular ML libraries in MLflow model format, they do not cover every use case. For example, you may 
+want to use a model from an ML library that is not explicitly supported by MLflow's built-in 
+flavors. Alternatively, you may want to package arbitrary inference code and data to create an 
+MLflow model. Fortunately, MLflow provides two solutions that can be used to accomplish these 
+tasks: :ref:`custom-python-models` and :ref:`custom-flavors`.
+
+.. _custom-python-models:
+
+Custom Python Models
+^^^^^^^^^^^^^^^^^^^^
+The :py:mod:`mlflow.pyfunc` module provides :py:func:`save_model() <mlflow.pyfunc.save_model>` and 
+:py:func:`log_model() <mlflow.pyfunc.log_model>` utilities for creating MLflow models with the
+``python_function`` flavor that contain arbitrary user-specified code and *artifact* (file) 
+dependencies. These artifact dependencies may include serialized models produced by any Python ML 
+library.
+
+Because these custom models contain the ``python_function`` flavor, they can be deployed
+to any of MLflow's supported production environments, such as SageMaker, AzureML, or local
+REST endpoints.
+
+The following examples demonstrate how the :py:mod:`mlflow.pyfunc` module can be used to create 
+custom Python models. For additional information about model customization with MLflow's 
+``python_function`` utilities, see the 
+:ref:`python_function custom models documentation <pyfunc-create-custom>`.
+
+Example: Creating a custom "add n" model
+****************************************
+In this example, we first define a class for a custom model that adds a specified numeric value, 
+`n`, to all columns of a Pandas DataFrame input. Then, we leverage the :py:mod:`mlflow.pyfunc` APIs 
+to save an instance of this model with `n = 5` in MLflow model format. Finally, we load the model in 
+``python_function`` format and use it to evaluate a sample input
+
+.. code:: python
+
+    import mlflow.pyfunc
+
+    # Define the model class
+    class AddN(mlflow.pyfunc.PythonModel):
+
+        def __init__(self, n):
+            self.n = n
+
+        def predict(self, context, model_input):
+            return model_input.apply(lambda column: column + self.n)
+
+    # Construct and save the model
+    model_path = "add_n_model"
+    add5_model = AddN(n=5)
+    mlflow.pyfunc.save_model(dst_path=model_path, python_model=add5_model)
+
+    # Load the model in `python_function` format
+    loaded_model = mlflow.pyfunc.load_pyfunc(model_path)
+
+    # Evaluate the model
+    import pandas as pd
+    model_input = pd.DataFrame([range(10)])
+    model_output = loaded_model.predict(model_input)
+    assert model_output.equals(pd.DataFrame([range(5, 15)]))
+
+
+Example: Saving an XGBoost model in MLflow format
+*************************************************
+In this example, we begin by training and saving a gradient boosted tree model using the XGBoost
+library. Next, we define a wrapper class around the XGBoost model that conforms to MLflow's
```

Comment:
In this example, we begin by training and saving a gradient boosted tree model using the XGBoost
library.  -> This example begins by training and saving a gradient boosted tree model using the XGBoost
library. 

Next, we define a wrapper class -> Next, it defines a wrapper class

Finally, we load the MLflow model  ... and use it -> Finally, it loads the MLflow model ... and uses it

Reply:
Fixed!